src/Api/Orbit.cpp

```diff
@@ -6,21 +6,21 @@
 #include "Api/Orbit.h"
 
 #include "Api/EncodedEvent.h"
+#include "Api/LockFreeApiEventProducer.h"
 #include "OrbitBase/Profiling.h"
 #include "OrbitBase/ThreadUtils.h"
 
-// Temporary dummy function, proper implementation is in next PR.
-// See https://github.com/google/orbit/pull/1702 for fully working prototype.
-static void EnqueueIntermediateEvent(const orbit_api::ApiEvent& event) { (void)event; }
-
 static void EnqueueApiEvent(orbit_api::EventType type, const char* name = nullptr,
                             uint64_t data = 0, orbit_api_color color = kOrbitColorAuto) {
-  uint64_t timestamp_ns = orbit_base::CaptureTimestampNs();
+  static orbit_api::LockFreeApiEventProducer producer;
```

Comment:
Just to clarify that I got it right here:
This will compiled into orbit.so.
The game will load orbit.so as soon as it is available.
In this case we are creating a static producer, which thus will be destructed on unload of orbit.so (or game exit).

Is there any issue with this?

Reply:
Yeah that's right, it should be very lightweight when we are not capturing, I think it's fine. But as mentioned above, one thing I'd like to add an api method to unload liborbit.so manually, although it might be tricky to implement.

Reply:
Okay, thanks for clarification